### ISSUE TITLE
Confusing thread UI with side replies

### ISSUE THREAD

**Issue by NelsonMinar** [cheeaun/phanpy#295]
Hey there, phanpy is fantastic. You have a very good taste for simplicity in UI, I really like the choices you make in this client.

One tiny thing I think is a bug: I got confused by the vertical bar showing thread responses in the Notifications display. 

![chrome_u5E2I7wMGY](https://github.com/cheeaun/phanpy/assets/382136/add3861f-c4f9-4007-a1f2-6279f1b808fc)

mhoye posted a megathread, replying only to himself. I've started the screenshot at message 9/X of his. I (nelson) responded to message 7/X, but Phanpy makes it look like it's a reply to 9/X. Then posted a 10/X replying to his 9/X. But phanpy put a vertical bar between my message and his 10/X, making it look like his 10/X was a direct reply to mine. It wasn't, it's a direct reply to 9/X.

Here's a [link to his 7/X post](https://mastodon.social/@mhoye/111336097458684403). Note that I deleted and reposted my reply after taking this screenshot, so what's on the public timeline now is slightly different from this screenshot. I think it's still confusing though.

My guess is the logic to display threads is doing something a little wrong with replies to long threads. It gets a little confused as to what is replying to what. This data is all pretty tricky, I'm not sure how to sort it out other than a bunch of test cases.

**Comment by cheeaun** [comment]
@NelsonMinar your screenshot seems cropped; want to confirm, is the 7/X post displayed above the 9/X post (with 8/X post)?

And you're right about the logic, it gets complicated to sort deep-nested replies, while grouping context by same-author threads) on one single line of posts 🙈 (That's why Phanpy has nested UI in the Post details page, but not for grouped timeline items). Yes, I need to write test cases 😂

**Comment by NelsonMinar** [comment]
You're right, my screenshot is missing the 7/X post. I cropped it wrong and now I can't reproduce it because the state has changed so much. Very sorry!

**Comment by runevision** [comment]
This is not only an issue with threads (author replying to themself) but with replies in general.

In this screenshot, the second post is a reply (by a second person) to the first and the third post (by a third person) is also a reply to the first post. However, the line makes it look like the third post is a reply to the second post.
![ReplyLineIssue](https://github.com/cheeaun/phanpy/assets/6685642/dcc96368-420e-474f-a9e1-8f276cced338)

IMO if a post displayed below another post is not a direct reply to that post, there should not be an unbroken line displayed between them.

In the screenshot above, the second post doesn't have any replies at all. In this case, IMO, there should not be a line continuing down from that post at all (unbroken or broken). However, a partial line could go upwards from the third post which stops or fades out at some point, without looking like it's connected to the second post.

**Comment by NelsonMinar** [comment]
Goodness my initial bug report is a mess, sorry. But since we're here, I just took a new screenshot showing the same problem, or a similar one.

The initial post by me @nelson is [here](https://tech.lgbt/@nelson/111964773219327603). There's a [reply to my original post by @sgillies](https://mastodon.social/@sgillies/111964794766247309) that I [replied to](https://tech.lgbt/@nelson/111964798908704005), then a second [reply to my original post by @gnomon](https://mastodon.social/@gnomon/111964803865425591) with no reply from me. So 4 posts total. One chain Original → @sgillies → @nelson and a second chain Original → @gnomon.

The Phanpy sidebar view displays this threading correctly.

![ohanpy sidebar centered on first post](https://github.com/cheeaun/phanpy/assets/382136/27277124-4393-4321-bef5-98fd9df34fd2)

But my Phanpy home timeline displays this incorrectly, as if @gnomon's reply was not to my original post but instead to my second post replying to @sgillies.

![phanpy home timeline](https://github.com/cheeaun/phanpy/assets/382136/77420fe6-8648-4ad5-8fa1-7e995b16122d)

I imagine all this reply ordering and timeline creation is quite complicated in the code! It's not a terrible bug. But at least I've got a clearer example. I'll endeavor not to delete the posts in question this time so you can examine them later.

**Comment by runevision** [comment]
I think I actually misinterpreted this issue originally and thought it was about the main timeline, but I see now the original screenshot is actually from the sidebar and the title indicates this as well.

I saw a bug in the sidebar replies today as well. In the screenshot below, the third post is indicated as being a reply to the second, but it's actually a reply to the first. The third post was posted "3 hours ago", while the second was posted "now", so that makes it extra clear they couldn't be connected the way the UI indicates they are.
![SideviewBug](https://github.com/cheeaun/phanpy/assets/6685642/0b995287-fe17-4cc3-9559-8b05e0a16f27)

**Comment by TheDecryptor** [comment]
Think I just hit this, numbered the replies in the screenshot since the censoring is a bit visually confusing.

1 is a comment on the first post, 2 is a reply to that comment, and 3 is a reply to 2. The threading indicators are correct, but the "parenting" is just wrong.

<img width="641" height="858" alt="Image" src="https://github.com/user-attachments/assets/80cfbd98-556d-4b24-ae8c-390ed26f20b5" />

If I focus on the comment, or any of the replies, the thread is shown as you'd expect. The issue only happens when viewing the root parent post.

**Comment by cheeaun** [comment]
@TheDecryptor that's probably a different bug. Can you DM me the post link? (cheeaun(at)mastodon.social)
